File: GJE.rs
fn main () {
    let mut m = [[8.0, -8.0, -9.0, -1.0], [-10.0, 15.0, -9.0, -25.0], [-9.0, -1.0, 7.0, 3.0]];
    
    if m[0][0] == 0.0 {
	    for i in 0..m[0].len() {
	        m[0][i] += 1.0;
	    }
    }
    
    for c in 0..m.len() {
	    if m[c][c] != 1.0 {
		    for e in 0..m[c].len() {
		        m[c][e] = m[c][e] / m[c][c];
		    }
	    }
        for r in 0..m.len() {
    		if r != c {
			for e in 0..m[r].len() {
			    m[r][e] -= m[c][e] * m[r][c]
			}
        	}
        }
    }
    
    for element in m {
        for element_ in element {
            print!("{}", element_);
        }
        print!("\n");
    }
}

File: GJE.py
#Python equivalent the Rust version was transcoded from by me

import numpy as np

M = np.array([[8, -8, -9, -1], [-10, 15, -9, -25], [-9, -1, 7, 3]], float)

if M[0][0] == 0:
    M[0] += 1

for j in range(len(M)):  # For each column
    if M[j][j] != 1:  # If the jth element of that column is not 1
        M[j] = M[j] / M[j][j]  # Then divide the row by that element

    for i in range(len(M)):  # For each row
        if i != j:  # If we are not at the row we want to have the pivot 1
            M[i] -= M[j] * M[i][j]

np.set_printoptions(precision=20)
display(M[:, -1])
